### ISSUE TITLE
Rack::Request::parse_multipart throws undefined method 'extract_multipart'

### ISSUE THREAD

**Issue by xfox64x** [rack/rack#1315]
I don't really know anything about Ruby or rack or how dependencies work, but another repo I like (beefproject) uses this rack thing, and it's been having some trouble. The trace produced is something like this:

```
E, [2018-10-01T11:42:38.684765 #8899] ERROR -- : Unexpected error while processing request: undefined method `extract_multipart' for Rack::Multipart:Module
	/var/lib/gems/2.5.0/gems/rack-2.0.6/lib/rack/request.rb:472:in `parse_multipart'
	/var/lib/gems/2.5.0/gems/rack-2.0.6/lib/rack/request.rb:335:in `POST'
	/var/lib/gems/2.5.0/gems/rack-2.0.6/lib/rack/request.rb:358:in `params'
	/var/lib/gems/2.5.0/gems/rack-2.0.6/lib/rack/request.rb:22:in `params'
        ...
```
As I've already stated: I don't really understand Ruby or how dependencies work in Ruby. But it would seem that adding the following to the requires in ```rack/requests.rb``` fixes the above issue: ```require 'rack/multipart'```

If anyone can shed some light on this, I would greatly appreciate it. All criticisms and spontaneous educations welcome.

**Comment by mikegee** [comment]
That's strange because `Rack::Multipart` [does define `extract_multipart`](https://github.com/rack/rack/blob/cb1fdb600bc525258b3c34ea95f1598ee6def9c6/lib/rack/multipart.rb#L43). Can you confirm that you are using the released gem at version 2.0.6, and not a git branch?

**Comment by jeremyevans** [comment]
My guess as to a possible cause of this issue is that `Rack::Multipart` is defined by some other code.  Since the module is already defined, the autoload that rack defines doesn't trigger. 

The fix in this case would be to require rack before attempting to define `Rack::Multipart`, so you should try that and see if it works.

I think the issue with requiring rack/multipart in rack/request is that it loads rack/multipart in cases where it is not needed.  That is what the autoload attempts to avoid.  FWIW, I'm against the use of autoload, but that ship sailed with rack long ago.

If you think this is still a bug in the master branch of rack, please submit a failing test or self contained example showing the error.
